unstructured/ingest/connector/chroma.py

```diff
@@ -0,0 +1,162 @@
+
+    # from chromadb.config import Settings as ChromaSettings
+
+
+@dataclass
+class SimpleChromaConfig(BaseConnectorConfig):
+    collection_name: str
+    path: t.Optional[str] = None
+    settings: t.Optional[t.Dict[str, str]] = None
+    tenant: t.Optional[str] = None
+    database: t.Optional[str] = None
+    host: t.Optional[str] = None
+    port: t.Optional[int] = None
+    ssl: bool = False
+    headers: t.Optional[t.Dict[str, str]] = ({},)
+
+
+@dataclass
+class ChromaWriteConfig(WriteConfig):
+    batch_size: int = 100
+
+
+@dataclass
+class ChromaDestinationConnector(BaseDestinationConnector):
+    write_config: ChromaWriteConfig
+    connector_config: SimpleChromaConfig
+    _collection: t.Optional["ChromaCollection"] = None
+
+    @property
+    def chroma_collection(self):
+        if self._collection is None:
+            self._collection = self.create_collection()
+        return self._collection
+
+    def initialize(self):
+        pass
+
+    @DestinationConnectionError.wrap
+    def check_connection(self):
+        _ = self.chroma_collection
+
+    @requires_dependencies(["chromadb"], extras="chroma")
+    def create_collection(self) -> "ChromaCollection":
+        import chromadb
+
+        if self.connector_config.path:
+            chroma_client = chromadb.PersistentClient(
+                path=self.connector_config.path,
+                settings=self.connector_config.settings,
+                tenant=self.connector_config.tenant,
+                database=self.connector_config.database,
+            )
+
+        elif self.connector_config.host and self.connector_config.port:
+            chroma_client = chromadb.HttpClient(
+                host=self.connector_config.host,
+                port=self.connector_config.port,
+                ssl=self.connector_config.ssl,
+                headers=self.connector_config.headers,
+                settings=self.connector_config.settings,
+                tenant=self.connector_config.tenant,
+                database=self.connector_config.database,
+            )
+        else:
+            raise ValueError("Chroma connector requires either path or host and port to be set.")
+
+        collection = chroma_client.get_or_create_collection(
+            name=self.connector_config.collection_name
+        )
+        return collection
+
+    @DestinationConnectionError.wrap
+    @requires_dependencies(["chromadb"], extras="chroma")
+    def upsert_batch(self, batch):
+        collection = self.chroma_collection
+
+        try:
+            # Chroma wants lists even if there is only one element
+            collection.add(
+                ids=batch["ids"],
+                documents=batch["documents"],
+                embeddings=batch["embeddings"],
+                metadatas=batch["metadatas"],
+            )
+        except Exception as e:
+            raise ValueError(f"chroma error: {e}") from e
+
+    @staticmethod
+    def chunks(iterable, batch_size=100):
```

Comment:
This seems usable for other connectors, maybe we can move it to ingest utils. I also had [something similar](https://github.com/Unstructured-IO/unstructured/pull/2152/files#diff-f9e7c943e53c317768eadafb884c26862d85943655348d0a56e17ec7b76be445R191) for elasticsearch.

Reply:
Agree. It should be "very" similar.
I think Chunks is too similar to Chunking though. We should rename it.